apps/server/src/utils/getFileListFromFolder.ts

```diff
@@ -1,9 +1,23 @@
 import { ProjectFile } from 'ontime-types';
 
 import { readdirSync, statSync } from 'fs';
+import { extname } from 'path';
+
+interface Options {
+  allowedExtensions?: Array<string>;
+}
+
+export const getFileListFromFolder = (folderPath: string, options: Options = {}): Array<ProjectFile> => {
+  let files = readdirSync(folderPath);
+
+  // Filter files if allowedExtensions is provided
+  if (options.allowedExtensions && options.allowedExtensions.length > 0) {
+    files = files.filter((file) => {
+      const ext = extname(file).toLowerCase();
+      return options.allowedExtensions.includes(ext);
+    });
+  }
```

Comment:
I might be missing something, but would we not return undefined if the options are not provided?

Would it be best to simplify this to our use case?
I would think about two functions
```
function getFilesFromFolder() // all the node file system stuff
function filterProjectFiles() // the actual logic here
```

we could do unit tests on the second one and potentially generalise it to filter any file

What do you think?

Reply:
my first idea was to have a function that we could pass the extensions argument, and if we didn't we'd just return everything

but I feel your idea is better, it's cleaner and also makes it easier to test. We could keep `getFilesFromFolder` private to the module for now, and only expose `filterProjectFiles`, what do you think?

Reply:
sounds good to me.
I guess in that case we might need two private functions `getFilesFromFolder` and `filterProjectFiles`
which we expose as `getProjectFiles`

But I am sure you will find a good way to do it. We just dont want to mess around with mocking too much and is better to extract the business logic from the accidental logic